documents/architecture-decisions.md

```diff
@@ -612,17 +625,147 @@ available as a template. An API will list all available templates to the user.
+   - each saved state gets a new version number,
+   - depending on the type of change made → the last version before 1.0.0-1 could
+   be 0.832.12.
+
+2. after the transition from approved → draft:
+   - here the "draft counter" is
+     incremented by 1 and remains the same during the whole processing cycle until
+     the next approval
+   - In this case, the changes that are then made in the "draft" state should be
+     traceable via a change index of the draft counter
+
+   - 1.0.0-1.0 becomes 1.0.0-1.1 with the first change in the draft state and
+     increases with each further saved change up to 1.0.0-1.X
+
+3. after the transition from published → draft, i.e. for a new version > 1.0.0:
+   - It is checks what type of change was made compared to thet las published
+     version and increments major, minor, or patch by 1 accordingly, however a
+     higher level change resets the lower level change back to 0 and each level
+     can be incremented by 1 at most
+     → 1.0.0 can become only: 2.0.0, 1.1.0, 1.0.1
+
+   - Here we will add the "draft counter" immediately when creating the new "draft"
+     version (-1) and thereafter increment the draft index with each save
+     (analog 2) )
+
+for versions > 1.0.0
+Here we will add the "draft counter" immediately when creating the new "draft"
+version (-1) and thereafter increment the draft index with each save (analog 2) )
+
+Regardless of the number of changes that theoretically led to the increase of
+the version number on each level.
+We would thus generate the potential new version number resulting from each
+change each time we save.
+
+##### Integer versioning
+
+![workflow](workflowIntegerVersioning.drawio.svg)
+
+With integer versioning, the version number is always incremented only when a
+new "draft" version is initially created (Predicted new version = target version).
+Exception: Version 0, until the first time "published".
+
+##### Konfiguration
+
+There should be a configuration that defines which type of versioning should be
+taken for new documents.
+For existing documents the ype of versioning can NOT change during lifetime,
+i.e. once Semantic Versioning,
+always Semantic Versioning (or analog for Integer Versioning).
+
+##### current_release_date
+
+When saving, it is always checked whether the current_release_date is in the
+past. In this case the date is set
+to the current date. In all other cases (date in the future) this remains.
+This does not affect the "preassignment" of the current_release_date when
+the "RfPublication" status is set.
+
+##### revision_history
+
+The 'tracking/revision_history' should be maintained in the backend
+
+- Semantic Versioning
+
+  - Pre 1.0.0
+    - When saving (both new creation and "normal" saving) a summary and legacy
+      version can be specified in a  modal window.
+    - The backend creates a new revision history element for each change with the
+        `current date`, the `summary` and the `legacy version` of the user.
```

Comment:
I don't completely get that sentence. Does it mean that the `date` is set to the current date (timestamp) and `summary` and `legacy_version` are given by the user?